frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala

```diff
@@ -331,6 +331,57 @@ class BspMetalsClientSpec(
     }
   }
 
+  test("compile producing Semanticdb with scala3 without using plugin") {
+    TestUtil.withinWorkspace { workspace =>
+      val `A` = TestProject(
+        workspace,
+        "A",
+        dummyFooSources,
+        scalaVersion = Some("0.23.0-bin-20200210-0dc07b0-NIGHTLY"),
+        scalaOrg = Some("ch.epfl.lamp"),
+        scalaCompiler = Some("dotty-compiler_0.23")
+      )
+      val projects = List(`A`)
+      val configDir = TestProject.populateWorkspace(workspace, projects)
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      WorkspaceSettings.writeToFile(
+        configDir,
+        WorkspaceSettings("4.3.0", List()),
+        logger
+      )
+      loadBspState(workspace, projects, logger) { state =>
+        val compiledState = state.compile(`A`).toTestState
+        assertExitStatus(compiledState, ExitStatus.Ok)
+        assertValidCompilationState(compiledState, projects)
+        assertSemanticdbFileFor("Foo.scala", compiledState)
+        assertSuccessfulCompilation(compiledState, projects, isNoOp = false)
+      }
+    }
+  }
+
+  test("compile not producing Semanticdb with scala3 when settings are not present") {
+    TestUtil.withinWorkspace { workspace =>
+      val `A` = TestProject(
+        workspace,
+        "A",
+        dummyFooSources,
+        scalaVersion = Some("0.23.0-bin-20200210-0dc07b0-NIGHTLY"),
+        scalaOrg = Some("ch.epfl.lamp"),
+        scalaCompiler = Some("dotty-compiler_0.23")
+      )
+      val projects = List(`A`)
+      val configDir = TestProject.populateWorkspace(workspace, projects)
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      loadBspState(workspace, projects, logger) { state =>
+        val compiledState = state.compile(`A`).toTestState
+        assertExitStatus(compiledState, ExitStatus.Ok)
+        assertValidCompilationState(compiledState, projects)
```

Comment:
Re my previous comments, these two assertions are enough :smile: It's important that we assert the compilation state is valid and it's a successful compilation because there are lots of invariants that need to be met for a compilation to be correct. 👍